apps/api/v2/src/modules/organizations/repositories/attributes/organization-attribute-option.repository.ts

```diff
@@ -0,0 +1,139 @@
+    private readonly dbWrite: PrismaWriteService,
+    private readonly organizationsAttributesService: OrganizationAttributesService
+  ) {}
+
+  async createOrganizationAttributeOption(
+    organizationId: number,
+    attributeId: string,
+    data: CreateOrganizationAttributeOptionInput
+  ) {
+    return this.dbWrite.prisma.attributeOption.create({
+      data: {
+        ...data,
+        attributeId,
+      },
+    });
+  }
+
+  async deleteOrganizationAttributeOption(organizationId: number, attributeId: string, optionId: string) {
+    return this.dbWrite.prisma.attributeOption.delete({
+      where: {
+        id: optionId,
+        attributeId,
+      },
+    });
+  }
+
+  async updateOrganizationAttributeOption(
+    organizationId: number,
+    attributeId: string,
+    optionId: string,
+    data: UpdateOrganizationAttributeOptionInput
+  ) {
+    return this.dbWrite.prisma.attributeOption.update({
+      where: {
+        id: optionId,
+        attributeId,
+      },
+      data,
+    });
+  }
+
+  async getOrganizationAttributeOptions(organizationId: number, attributeId: string) {
+    return this.dbRead.prisma.attributeOption.findMany({
+      where: {
+        attributeId,
+      },
+    });
+  }
+
+  async getOrganizationAttributeOptionsForUser(organizationId: number, userId: number) {
+    const options = this.dbRead.prisma.attributeOption.findMany({
+      where: {
+        attribute: {
+          teamId: organizationId,
+        },
+        assignedUsers: {
+          some: {
+            member: {
+              userId,
+            },
+          },
+        },
+      },
+    });
+
+    return options;
+  }
+  async assignOrganizationAttributeOptionToUser(
+    organizationId: number,
+    userId: number,
+    data: AssignOrganizationAttributeOptionToUserInput
+  ) {
+    const attribute = await this.organizationsAttributesService.getOrganizationAttribute(
+      organizationId,
+      data.attributeId
+    );
+    if (!attribute) throw new Error("Attribute not found");
+
+    if (!TYPE_SUPPORTS_VALUE.has(attribute.type) && data.value) {
+      throw new Error("Attribute type does not support value");
+    }
+
+    const membership = await this.dbRead.prisma.membership.findFirst({
+      where: { teamId: organizationId, userId, accepted: true },
+    });
+
+    if (!membership) throw new Error("Membership not found");
+
+    let attributeOptionId = data.attributeOptionId;
+    if (data.value && !attributeOptionId) {
+      const attributeOption = await this.createOrganizationAttributeOption(organizationId, attribute.id, {
+        value: data.value,
+        slug: slugify(data.value),
+      });
+      attributeOptionId = attributeOption.id;
+    }
+
+    if (!attributeOptionId) throw new Error("Attribute option not found");
+
+    return this.dbWrite.prisma.attributeToUser.create({
+      data: {
+        attributeOptionId,
+        memberId: membership.id,
+      },
+    });
+  }
```

Comment:
we should limit the amount of logic within repository and instead have single responsibility functions chained within services, error handling can also be in the services